### PR TITLE
feat: move cache control policy to standalone config section

### PR DESCRIPTION
**Pull request by df-wg**
In https://github.com/wundergraph/cosmo/pull/1155, we added in support for having a restrictive cache control policy, as part of the header rules. This pulls out the cache control configuration into a standalone section of the configuration, for clarity and extensibility. 

It can now be configured like: 
```yaml

cache_control_policy:
  enabled: true
  value: "max-age=180, public"
  subgraphs:
    - name: "products"
      value: "max-age=60, public"
```

Additionally, in https://github.com/wundergraph/cosmo/pull/1196 we had created a `set` header operation. This PR adds tests to validate that using the `set` header operation, we can, if necessary, override the cache control restrictions, as a 'break-glass' measure. 

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: router/core/header_rule_engine.go [hunk 1]
```diff
@@ -136,6 +138,31 @@ func NewHeaderPropagation(rules *config.HeaderRules) (*HeaderPropagation, error)
 	return &hf, nil
 }
 
+func (hf *HeaderPropagation) convertCacheToRules(cacheControl config.CacheControlPolicy) {
```
Comment (df-wg) [inline router/core/header_rule_engine.go:141]: This is the direction I found simplest - to simply convert the cache control policy object into the appropriate rules. It isn't the only way, and I'm happy to get push-back on it, I had just thought it was the simplest and fastest way of adding in the functionality

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/core/header_rule_engine.go [hunk 2]
```diff
@@ -136,6 +138,31 @@ func NewHeaderPropagation(rules *config.HeaderRules) (*HeaderPropagation, error)
...
+	if cacheControl.Enabled {
+		hf.rules.All.Response = append(hf.rules.All.Response, &config.ResponseHeaderRule{
+			Operation: config.HeaderRuleOperationPropagate,
+			Algorithm: config.ResponseHeaderRuleAlgorithmMostRestrictiveCacheControl,
+			Default:   cacheControl.Value,
+		})
+	}
+
+	for _, graph := range cacheControl.Subgraphs {
```
[the remark this answers is not in the document]
Comment (df-wg) [reply router/core/header_rule_engine.go:152]: I don't think there's a need fo rthat. It's from their config, and if their config is flawed, do we handle that for them?

[the remark this answers is not in the document]
Comment (StarpTech) [reply router/core/header_rule_engine.go:159]: We could add a warning but I'm good to go with it.

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**